src/crewai/agent.py

```diff
@@ -261,6 +262,9 @@ def execute_task(
                 }
             )["output"]
         except Exception as e:
+            if isinstance(e, LiteLLMAuthenticationError):
+                # Do not retry on authentication errors
+                raise e
```

Comment:
nice ! When we raise should we create a colored logger to make it clear no keys provided ?